plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/ServerSettingsWizardPageModel.java

```diff
@@ -362,9 +362,9 @@ private String getProfileId() {
 	}
 
 	private String getRoute(boolean isDefaultRoute, IRoute route) {
-		if (!isDefaultRoute || route == null) {
+		// We must return a route here, because setting the server host depends on the route
```

Comment:
This just removes the feature, which allows user to select route later at starting Show In Browser. Does server adapter logic allow for such a feature? If it does, then the solution may be just to fill host with empty string or a message reminding user of his choice to choose always (or until he selects 'remember' option) the route in a dialog.